### ISSUE TITLE
Customer gets changed when FreeScout receives auto-reply from Mail.Ru mailboxes

### ISSUE THREAD

**Issue by ilyakonovalenko** [freescout-helpdesk/freescout#2826]
Hello,

I noticed FreeScout changes the customer when receives auto-reply generated from Mail.Ru mailboxes.

Headers are the following (mailbox name is replaced by $MAILBOX):
```
From: $MAILBOX@mail.ru
Auto-Submitted: auto-replied
X-AutoReply: $MAILBOX@mail.ru
Precedence: bulk
Reply-To: $MAILBOX.autoreply@mail.ru
Errors-To: $MAILBOX.autoreply@mail.ru
```

Can this be fixed please.

**Comment by freescout-helpdesk** [comment]
"Customer gets changed" - what does it mean?

**Comment by ilyakonovalenko** [comment]
I meant FreeScout changes customer for conversation from $MAILBOX@mail.ru to $MAILBOX.autoreply@mail.ru (example, the same is for any mail.ru domains).

**Comment by freescout-helpdesk** [comment]
`Reply-To` header in FreeScout has precedence over `From`. So it looks like there is no a simple solution for this.

**Comment by ilyakonovalenko** [comment]
Maybe just skip any changes in case the email has header Auto-Submitted: auto-replied?

**Comment by freescout-helpdesk** [comment]
Instruction on how to reproduce the issue is needed.

**Comment by ilyakonovalenko** [comment]
Here are steps:
1) Send an email from any mailbox on mail.ru email service with configured auto-reply.
This is how the conversation looks like in FreeScout after the initial message:
![изображение_2023-03-15_160038619](https://user-images.githubusercontent.com/63669538/225331928-36211dcb-b3d3-4b7c-b5f0-fdad24f9ac05.png)
2) Send a reply from FreeScout in that conversation.
3) Mail.ru server will then generate auto-reply and this is how the conversation looks like now:
![изображение_2023-03-15_160343872](https://user-images.githubusercontent.com/63669538/225332844-bcb2a7ca-fc37-4856-93e3-45a07a5c6d23.png)

**Comment by freescout-helpdesk** [comment]
We've made some changes in the `master` branch. Feel free to check.

**Comment by jamieppuk** [comment]
this update seems to have messed up our incoming tickets
so when someone fills in a contact form on our website it uses our ouremail@ourdomain.co.uk email smtp settings to send the email to itself which is then picked up by freescout

Typical headers would be...
Delivered-To:	ouremail@ourdomain.co.uk
Return-Path:	<ouremail@ourdomain.co.uk>
Delivered-To:	ouremail@ourdomain.co.uk
Reply-To: Mister Customer <customer@gmail.com>

Before the update this would correctly create the ticket in the customer customer@gmail.com account with replies to the ticket going to Mister Customer <customer@gmail.com>
After the update the Reply-To: name and address is ignored and they are instead created in a ouremail@ourdomain.co.uk account.
Is this an oversight that can be corrected or is this now how it will be and we will need to update our system to work with it ?
Thanks

**Comment by freescout-helpdesk** [comment]
FYI. We had to revert this change as it creates problems in other cases: https://github.com/freescout-helpdesk/freescout/issues/2833

So for this particular case there is no a solution for now.

**Comment by ilyakonovalenko** [comment]
Have you found what exactly caused the issue mentioned in #2833? I will try to find the solution which will work for both of us.

**Comment by ilyakonovalenko** [comment]
Okay, according to the code:
```
$autoresponder_headers = [
            'x-autoreply'    => '',
            'x-autorespond'  => '',
            'auto-submitted' => '', // this can be auto-replied, auto-generated, etc.
            'precedence' => ['auto_reply', 'bulk', 'junk'],
            'x-precedence' => ['auto_reply', 'bulk', 'junk'],
        ];
```
The issue mentioned in #2833 is that there is a header `Precedence: junk`. So technically your fix for my issue is correct, because "junk" should not be set (normally).

**Comment by ilyakonovalenko** [comment]
@freescout-helpdesk Can we please have the issue re-opened so that community can discuss and find a solution?

**Comment by ilyakonovalenko** [comment]
I tested various scenarios and found that your fix from 1.8.68 was correct and @jamieppuk problem was `junk` flag in `\MailHelper::isAutoResponder`. The flag does not mean the email is auto-responder, so it should probably be removed from the `$autoresponder_headers`.

Not sure though what was the reason to add `junk` which means the email is spam. If it is needed for some other stuff, please let me know so that I can look for another solution.
If it is not really needed, then after removing `junk` from the array in `isAutoResponder()` both my issue (fix from 1.8.68) and @jamieppuk issue should be fixed.

**Comment by freescout-helpdesk** [comment]
We'll keep 'junk' in `$autoresponder_headers` as so far it works fine. For now we can only suggest to adjust the behaviour via a custom module.
